### PR TITLE
[paradisebotsapi.js] Add types

### PR DESCRIPTION
**Pull request by jzeuzs**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/paradisebotsapi.js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Type definitions for paradisebotsapi.js 1.0
+// Project: https://github.com/ParadiseBotList/paradisebotsapi.js#readme
+// Definitions by: Yoshida Tomio <https://github.com/1chiSensei>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare class PBL {
+    constructor(id?: string, auth?: string);
```
Comment (peterblazejewicz) [inline types/paradisebotsapi.js/index.d.ts:7]:
```suggestion
    constructor(id: string, auth: string);
```
https://github.com/ParadiseBotList/paradisebotsapi.js#arguments
those cannot be optional IMO, one is used in string template, other in authorization header.

Comment (peterblazejewicz) [reply types/paradisebotsapi.js/index.d.ts:7]: can you test this using runtime code? I can see README using parameterless constructor.
The git history shows those were required:
https://github.com/ParadiseBotList/paradisebotsapi.js/commit/6b64a5dd171368786cb010c09d73ea5a925daec8#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R5-R6

File: types/paradisebotsapi.js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+    post(server_count: number, shard_count: number): Promise<void>;
+    get(botID: string, response: (data: Record<string, unknown>) => void): Promise<void>;
+    getUser(userID: string, response: (data: Record<string, unknown>) => void): Promise<void>;
+}
+
+export = PBL;
```
Comment (amcasey) [inline types/paradisebotsapi.js/index.d.ts:14]: This export shape appears to be new in [1.0.6](https://github.com/ParadiseBotList/paradisebotsapi.js/commit/2bb64206e18ba3d62623e5188eb60a308ff1c3e6#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346), but the newest published version seems to be [1.0.1](https://unpkg.com/browse/paradisebotsapi.js@1.0.1/index.js).

File: types/paradisebotsapi.js/index.d.ts (hunk 1 above)
Comment (amcasey) [reply types/paradisebotsapi.js/index.d.ts:7]: Sensible question.

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/paradisebotsapi.js/index.d.ts (hunk 1 above)
Comment (1chiSensei) [reply types/paradisebotsapi.js/index.d.ts:7]: but you wont need this when running `.get()` and `.getUser()`

File: types/paradisebotsapi.js/index.d.ts (hunk 2 above)
Comment (1chiSensei) [reply types/paradisebotsapi.js/index.d.ts:14]: There are multiple packages [paradiseapi.js](https://npmjs.com/package/paradiseapi.js) and [paradisebotsapi.js](https://npmjs.com/package/paradisebotsapi.js)

Comment (1chiSensei) [reply types/paradisebotsapi.js/index.d.ts:14]: ill be changing it in a while i have classes

Comment (amcasey) [reply types/paradisebotsapi.js/index.d.ts:14]: No rush. 😄   I'm not sure I understand though - these appear to be the types for paradisebotsapi.js, not paradiseapi.js.

Comment (1chiSensei) [reply types/paradisebotsapi.js/index.d.ts:14]: Yep. But they have the same repo so I'll use unpkg so I can get the correct source.

File: types/paradisebotsapi.js/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/paradisebotsapi.js/index.d.ts:7]: no worries, it works, tested locally:
```js
new PBL()
PBL { id: undefined, auth: undefined }
```

File: types/paradisebotsapi.js/index.d.ts (hunk 2 above)
Comment (peterblazejewicz) [reply types/paradisebotsapi.js/index.d.ts:14]: my local NPM installs `"paradiseapi.js@1.10.0"`, not `1.0.1`

File: types/paradisebotsapi.js/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// Type definitions for paradisebotsapi.js 1.0
```
Comment (peterblazejewicz) [inline types/paradisebotsapi.js/index.d.ts:1]:
```suggestion
// Type definitions for paradisebotsapi.js 1.10
```
If this is for 1.10.0, this should be changed

File: types/paradisebotsapi.js/index.d.ts (hunk 2 above)
Comment (1chiSensei) [reply types/paradisebotsapi.js/index.d.ts:14]: Its `paradisebotsapi.js` not `paradiseapi.js`

**Review by amcasey — APPROVED** [review]
Thanks for straightening things out!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@1chiSensei thx!

**Comment by 1chiSensei** [comment]
Ready to merge

**Pull request merged**